### PR TITLE
fixed bugs UUID

### PR DESCRIPTION
**Pull request by piotr-major-rmtcfm-com**

### CODE REVIEW

File: app/controllers/ToDo/DestroyController.js [hunk 1]
```diff
@@ -16,15 +16,20 @@ class DestroyController {
...
         this.todoRepository = todoRepository;
+        this.isValidUuidServices = isValidUuidServices;
     }
 
     async invoke(request, response) {
         const {
             params: { id: todoId }
         } = request;
 
+        if (!(await this.isValidUuidServices.handle(todoId))) {
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/DestroyController.js:29]: await not needed

File: app/controllers/ToDo/PatchController.js [hunk 2]
```diff
@@ -32,6 +33,10 @@ class PatchController {
             params: { id: todoId }
         } = request;
 
+        if (!(await this.isValidUuidServices.handle(todoId))) {
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/PatchController.js:36]: as above, change it everywhere

File: app/controllers/User/ShowController.js [hunk 3]
```diff
@@ -15,12 +15,20 @@ class ShowController {
...
         this.userRepository = userRepository;
+        this.isValidUuidServices = isValidUuidServices;
     }
 
     async invoke(request, response) {
-        const user = await this.userRepository.findById(request.params.id);
+        const {
+            params: { id: userId }
+        } = request;
+        if (!(await this.isValidUuidServices.handle(userId))) {
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/User/ShowController.js:27]: empty line

File: app/controllers/User/UpdateController.js [hunk 4]
```diff
@@ -56,6 +57,10 @@ class UpdateController {
             params: { id: userId }
         } = request;
 
+        if (!(await this.isValidUuidServices.handle(request.params.id))) {
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/User/UpdateController.js:60]: handle(userId)

File: app/controllers/User/DestroyController.js [hunk 5]
```diff
@@ -16,11 +16,16 @@ class DestroyController {
...
      *   @apiError BadRequest
      */
-    constructor(userRepository) {
+    constructor(userRepository, isValidUuidServices) {
         this.userRepository = userRepository;
+        this.isValidUuidServices = isValidUuidServices;
     }
 
     async invoke(request, response) {
+        if (!(await this.isValidUuidServices.handle(request.params.id))) {
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/User/DestroyController.js:25]: save `request.params.id` to any variable

File: app/controllers/ToDo/DestroyController.js (hunk 1 above)
Comment (daniel-pupczyk-rmtcfm-com) [reply app/controllers/ToDo/DestroyController.js:29]: also call it `isUUIDValidHandler` everywhere

File: app/di/controllers/todo.js [hunk 6]
```diff
@@ -6,23 +6,23 @@ module.exports = {
...
         },
         'controllers.todo.updateController': {
             class: '/controllers/ToDo/UpdateController',
-            arguments: ['@repositories.todo']
+            arguments: ['@repositories.todo', '@services.IsValidUuid']
         },
         'controllers.todo.patchController': {
             class: '/controllers/ToDo/PatchController',
-            arguments: ['@repositories.todo']
+            arguments: ['@repositories.todo', '@services.IsValidUuid']
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/di/controllers/todo.js:25]: it should be lowercase `isUUIDValidHandler`

File: app/services/IsValidUuid.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+const { validate } = require('uuid');
+
+class isValidUuid {
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/services/IsValidUuid.js:3]: class IsUUIDValidHandler

File: app/controllers/ToDo/DestroyController.js (hunk 1 above)
Comment (rafal-ksiazek-rmtcfm-com) [reply app/controllers/ToDo/DestroyController.js:29]: Wait a minute. Checking params should be done in validators. 
Please not create services with method with one simple line. 
In validators it's one line: `validate(uuid);`
You created service: 6 lines, injected it several times, multiple changes. 

Validators, validators, validators!!!!!

File: app/controllers/ToDo/PatchController.js [hunk 8]
```diff
@@ -32,6 +33,10 @@ class PatchController {
...
+            return response.sendStatus(StatusCodes.NOT_FOUND);
```
Comment (rafal-ksiazek-rmtcfm-com) [inline app/controllers/ToDo/PatchController.js:37]: It's not NOT_FOUND, it's BAD_REQUEST error!!!!!

Comment (daniel-pupczyk-prpl) [reply app/controllers/ToDo/PatchController.js:37]: @rafal-ksiazek-rmtcfm-com  are you sure if there is 400? if user wants to test API and sends a request to `/users/something` then he receives 400 for any reason with a message that `UUID is not valid`, most of the users do not know what UUID is

**Pull request merged**